tests/testthat/test-forecasters-data.R

```diff
@@ -7,79 +7,105 @@ forecasters <- tibble::tribble(
   scaled_pop, list(1, TRUE), list("ahead", "pop_scaling"), "scaled_pop",
   scaled_pop, list(1, FALSE), list("ahead", "pop_scaling"), "scaled_pop",
   flatline_fc, list(1), list("ahead"), "flatline_fc",
-  smoothed_scaled, list(1, list(c(0,7,14), c(0))), list("ahead", "lags"), "smoothed_scaled"
+  smoothed_scaled, list(1, list(c(0, 7, 14), c(0))), list("ahead", "lags"), "smoothed_scaled"
 )
 expects_nonequal <- c("scaled_pop", "smoothed_scaled")
-synth_mean <- 25
-synth_sd <- 2
-tiny_sd <- 1.0e-5
-simple_dates <- seq(as.Date("2012-01-01"), by = "day", length.out = 40)
-# technically white noise, but with a variance that is miniscule
-constant <- epiprocess::as_epi_archive(tibble(
-  geo_value = "al",
-  time_value = simple_dates,
-  version = simple_dates,
-  a = synth_mean
-))
-
 # wrap a call that is made quite frequently
 # n_training_pad is set to avoid warnings from the trainer
-get_pred <- function(dataset,
-                     ii, outcome = "a", extra_sources = "") {
-  res <- slide_forecaster(
+default_slide_forecaster <- function(dataset,
+                                     ii, outcome = "a", extra_sources = "", linreg_warnings = TRUE) {
+  # linear_reg really doesn't like exactly equal data, and throws a warning.
+  # wrapperfun is to suppress that for forecasters using linear_reg
+  if (any(forecasters$fc_name[[ii]] %in% expects_nonequal) && linreg_warnings) {
```

Comment:
Shouldn't this be `!linreg_warnings`? Weird to say linreg_warnings = TRUE and have them get supressed.

Reply:
This is for cases where we know there are warnings, and that they don't matter and should be ignored.